server/src/main/java/com/alibaba/fescar/server/coordinator/DefaultCore.java

```diff
@@ -91,10 +91,12 @@ private GlobalSession assertGlobalSession(long transactionId, GlobalStatus statu
     }
 
     @Override
-    public void branchReport(BranchType branchType, String xid, long branchId, BranchStatus status, String applicationData) throws TransactionException {
-        GlobalSession globalSession = SessionHolder.findGlobalSession(XID.getTransactionId(xid));
+    public void branchReport(BranchType branchType, String resourceId, FragmentXID xid, long branchId,
```

Comment:
why need resourceId?

Reply:
the go server will check the resourceId was already registered